ddtrace/internal/_encoding.pyx

```diff
@@ -585,7 +585,7 @@ cdef class MsgpackEncoderV03(MsgpackEncoderBase):
                 if ret != 0:
                     return ret
                 # pack the value of a span link field (values can be number, string or dict)
-                if isinstance(v, Number):
+                if isinstance(v, NumericType):
```

Comment:
Maybe we could also test here for "int" to try to avoid that odd error? (isinstance can take a tuple of types to check for any of them).

Reply:
Yup. We already have test coverage for this: https://github.com/DataDog/dd-trace-py/blob/ffdea869d4e692517fa154396352becebd75519b/tests/tracer/test_encoders.py#L438.

isinstance(<some_int>, Number)` always returns `True` on my machine and in ci. There might be a cython or platform specific bug that causes this issue.